`ExtensionDescriptorMojoTest.shouldFlagMissingDependenciesInARealisticExtension` fails in Early Access JDK build

### Description

```
[WARN] Maven will be executed in interactive mode, but no input stream has been configured for this MavenInvoker instance.
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------< io.quackiverse:quarkus-fake-extension >----------------
[INFO] Building Quarkus - Fake extension - Deployment 999-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for io.quarkus:quarkus-core:jar:999-SNAPSHOT is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.157 s
[INFO] Finished at: 2023-03-01T01:23:49Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project quarkus-fake-extension: Could not resolve dependencies for project io.quackiverse:quarkus-fake-extension:jar:999-SNAPSHOT: Could not find artifact io.quarkus:quarkus-core:jar:999-SNAPSHOT -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[WARN] Maven will be executed in interactive mode, but no input stream has been configured for this MavenInvoker instance.
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------< io.quackiverse:quarkus-fake-extension-deployment >----------
[INFO] Building Quarkus - Fake extension - Deployment 999-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for io.quackiverse:quarkus-fake-extension:jar:999-SNAPSHOT is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.259 s
[INFO] Finished at: 2023-03-01T01:23:51Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project quarkus-fake-extension-deployment: Could not resolve dependencies for project io.quackiverse:quarkus-fake-extension-deployment:jar:999-SNAPSHOT: Could not find artifact io.quackiverse:quarkus-fake-extension:jar:999-SNAPSHOT -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] Tests run: 7, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 30.097 s <<< FAILURE! - in io.quarkus.maven.ExtensionDescriptorMojoTest
[ERROR] io.quarkus.maven.ExtensionDescriptorMojoTest.shouldFlagMissingDependenciesInARealisticExtension  Time elapsed: 3.528 s  <<< ERROR!
junit.framework.AssertionFailedError: 
Message format does not match expectations: 
Failed to resolve dependencies of io.quackiverse:quarkus-fake-extension:jar:999-SNAPSHOT
```
https://github.com/quarkusio/quarkus/actions/runs/4298664366/jobs/7493035836

> [ERROR] Failed to execute goal on project quarkus-fake-extension: Could not resolve dependencies for project io.quackiverse:quarkus-fake-extension:jar:999-SNAPSHOT: Could not find artifact io.quarkus:quarkus-core:jar:999-SNAPSHOT -> [Help 1]

Doesn't look right in terms of build reactor order. `extension-maven-plugin` is located under `independent-projects` and I don't think it should depend on core like that?
This issue does not pop up in regular CI because it builds everything initially, before running any tests.
In case this test has to stay like that we might need to adjust the EA build.

### Implementation ideas

_No response_

--- a/independent-projects/extension-maven-plugin/src/test/resources/fake-extension-runtime-with-missing-dependencies/pom.xml
+++ b/independent-projects/extension-maven-plugin/src/test/resources/fake-extension-runtime-with-missing-dependencies/pom.xml
@@ -17,7 +17,9 @@
     <dependency>
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-core</artifactId>
-      <version>${project.version}</version>
+      <!-- The version in this dependency is arbitrary, but needs to be 'real' (downloadable from maven).
+      It should not be a snapshot build, to avoid introducing a build-order dependency into this test. -->
+      <version>2.12.0.Final</version>
     </dependency>
 
     <!-- Mirror of the deployment artifact's dependencies would be here, but is deliberately missing -->

--- a/independent-projects/extension-maven-plugin/src/test/resources/fake-extension-runtime/pom.xml
+++ b/independent-projects/extension-maven-plugin/src/test/resources/fake-extension-runtime/pom.xml
@@ -19,7 +19,7 @@
   </properties>
 
   <!-- The versions in these dependencies are arbitrary, but need to match
-  in the runtime and deployment poms. -->
+  in the runtime and deployment poms. They should not be a snapshot build, to avoid introducing a build-order dependency into this test. -->
   <dependencies>
     <dependency>
       <groupId>io.quarkus</groupId>

